src/tools/miri/src/concurrency/weak_memory.rs

```diff
@@ -419,15 +406,15 @@ impl<'tcx> StoreBuffer {
             // In the language provided in the paper, an atomic store takes the value from a
             // non-atomic memory location.
             // But we already have the immediate value here so we don't need to do the memory
-            // access
-            val,
+            // access.
+            val: Some(val),
```

Comment:
Is there a reason to represent an uninit store buffer as one containing a store element with `None` val, instead of just an empty buffer with no elements?

Reply:
This is related to the new test I added: if the memory is uninitialized when the atomic object is created, an outdated read hitting that initial state should properly report UB.

Reply:
So we can initially have uninitialized memory, but we do not have a mechanism to write uninitialized bytes later?

Reply:
Yes, indeed. Currently all atomic intrinsics take `iN` or `*const T` types which must be initialized.

If we ever have `MaybeUninit` atomics, this will become a lot more tricky, since the value could be partially initialized.